js homework week3/MoreJavaScript.js

```diff
@@ -0,0 +1,211 @@
+// 1.Create a function that takes 3 arguments and returns the sum of the these arguments.
+function sum(a, b, c) 
+{
+return a + b + c
+};
+console.log(sum);
+sum(10, 22, 3);
+console.log(sum(10, 22, 3));
+
+//2.Create a function named colorCar that receives a color, and prints out, 'a red car' for example.
+
+function colorCar(color)
+{
+return 'a ' + color + ' car'
+};
+console.log(colorCar);
+colorCar('red');
+console.log(colorCar('red'));
+colorCar('blue');
+console.log(colorCar('blue'));
+
+//3.Create an object and a function that takes the object as a parameter and prints out all of its properties and values.
+
+const myObject = {shape: 'square' , width: 15 , height: 20};
+
+function funcObject (object) 
+{
+let property = Object.keys(object);
+let value = Object.values(object);
+    for (let i = 0; i < property.length; i++) 
+    {
+        console.log(property[i] + ': ' + value[i])
+    };
+
+}
+console.log(funcObject);
+funcObject(myObject);
+
+//4.Create a function named vehicleType that receives a color, and a code, 1 for car, 2 for motorbike. And prints 'a blue motorbike' for example when called as vehicleType("blue", 2)
+
+function vehicleType (color, code)
+{
+if (code === 1) { code = 'car'}
+else if (code === 2) {code = 'motorbike'};
+console.log('a ' + color + ' ' + code);
+};
+console.log(vehicleType);
+vehicleType('blue', 1);
+vehicleType('red', 2)
+
+//5.Can you write the following without the if statement, but with just as a single line with console.log(...);?
+
+3 === 3 ? console.log("yes") : console.log("no");
+
+//6.Create a function called vehicle, like before, but takes another parameter called age, so that vehicle("blue", 1, 5) prints 'a blue used car'
+
+function vehicle (color, code, age)
+{
+    if (code === 1) { code = 'car'}

```

Comment:
Ah this works, but I would advise not to modify `arguments` that you pass through the function. Because it can also lead to silly mistakes (which are sometimes very hard to find).

I would advise to create a separate variable where you store the `car` or `motorbike` in.